srcCpp03/RTIDDSImpl.h

```diff
@@ -38,7 +38,7 @@ class RTIDDSImpl : public IMessaging
 {
   public:
 
-    RTIDDSImpl();
+    RTIDDSImpl(bool isFlatData=false);
```

Comment:
1) Why do you pass isFlatData as a constructor argument, when all the other options are part of that `_PM` type.
2) If you still pass this argument, the constructor must be `explicit` to disable implicit conversion from `bool` to `RTIDDSImpl`.

Reply:
I have removed that argument and initialized it to _false_ in the constructor for `RTIDDSImpl` and to _true_ in `RTIDDSImpl_FlatData`.

```suggestion
template <typename T>
RTIDDSImpl<T>::RTIDDSImpl():
        _InstanceMaxCountReader(dds::core::LENGTH_UNLIMITED), //(-1)
        _InstanceHashBuckets(dds::core::LENGTH_UNLIMITED), //(-1)
        _isLargeData(false),
        _isFlatData(false),
        ...

...

template <typename T>
RTIDDSImpl_FlatData<T>::RTIDDSImpl_FlatData() {
    this->_isFlatData = true;
}
```